### ISSUE TITLE
No configureable controller input leads to bad inputs

### ISSUE THREAD

**Issue by sprockkets** [Mr-Wiseguy/Zelda64Recomp#508]
I have two linux setups I will report on. On my Opensuse 15.6 computer, I have a /dev/input/js0 input that shows up on boot to a USB device that doesn't exist. KDE can show that for this input, the control stick values for it list as -32767. When I use any controller, it will pair up to /dev/input/js1 or an event slot, and it works fine, and the js0 will then disappear.

However, it seems this program is grabbing every input possible and thus blending both the js0 and js1 controller, and what ends up happening, is the cursor continuously moves down, and to the left. I can push my control stick to counteract it, but obviously that only stops it from moving, and cannot then move to the right or up during game play.

What is weird, is that inputs work perfectly fine while in the control menus. I tried killing js0 by changing the permissions to be blocked by everyone, but it doesn't help.

On another computer, I run opensuse tumbleweed on a build from Nov 2024. It doesn't have this issue to begin with, so I'm guessing this is not an issue most run into.

In short, maybe this issue could be solved if 

1. Input is limited to a controller you can select OR
2. Maybe there is a way to specify this in the config files.

For my opensuse setup, I do use Steam, but it is closed and wasn't opened. I also made a new user account on the opensuse 15.6 system, and the same issue happens.

**Comment by Mr-Wiseguy** [comment]
I don't have any immediate plans to create a controller selection menu or a config file option for it. Even if I were to do it, it likely wouldn't happen for many months in all honesty with the current feature backlog I have. I'm also not a huge fan of having a menu to pick a controller for a single-player game; most native single-player PC games don't have such a menu so that would make recomp the odd one out. Even if recomp itself had a way to work around this, it'd still be a problem that would affect many more programs on your PC than just recomp so I think you'll have to figure out the issue that's causing js0 to produce bad inputs.

**Comment by sprockkets** [comment]
Hey, I figured out what the problem is. I have a H2O mini keyboard plugged in, and for whatever reason, not only presents itself with a USB id as a Vivitar camera, it also shows up as a controller. I removed it and now it's fine. Leaving this for anyone later to find if they have this issue.
